evm/src/cpu/kernel/asm/rlp/read_to_memory.asm

```diff
@@ -13,29 +13,26 @@ global read_rlp_to_memory:
 
     PUSH @SEGMENT_RLP_RAW // ctx == virt == 0
     // stack: addr, final_addr, retdest
-
 read_rlp_to_memory_loop:
     // stack: addr, final_addr, retdest
     DUP2
     DUP2
-    EQ
-    // stack: addr == final_addr, addr, final_addr, retdest
+    LT
+    ISZERO
+    // stack: addr >= final_addr, addr, final_addr, retdest
     %jumpi(read_rlp_to_memory_finish)
-    // stack: addr, len, retdest
-    DUP1
-    PROVER_INPUT(rlp)
-    // stack: byte, addr, addr, final_addr, retdest
-    MSTORE_GENERAL
     // stack: addr, final_addr, retdest
-    %increment
+    PROVER_INPUT(rlp)
+    SWAP1
+    MSTORE_32BYTES_32
```

Comment:
We need to make sure we don't have an empty blob (length 0), otherwise it would fail the CTL because of MSTORE_UNPACKING.